service/src/main/java/org/cbioportal/genome_nexus/service/internal/MyVariantInfoServiceImpl.java

```diff
@@ -65,7 +65,7 @@ public MyVariantInfo getMyVariantInfoByHgvsVariant(String variant)
             myVariantInfos = this.getMyVariantInfoByMyVariantInfoVariant(queryVariants);
             // manually set the original hgvs variant field
             for (MyVariantInfo myVariantInfo: myVariantInfos) {
-                myVariantInfo.setHgvs(queryToVariant.get(myVariantInfo.getQuery()));
+                myVariantInfo.setHgvs(queryToVariant.get(myVariantInfo.getVariant()));
```

Comment:
We need to make sure that My Variant Info returns a valid `variant` for all queries. As far as I remember query is our original input, variant is something returned by MVI.

Reply:
@onursumer the query field was actually null in most of the cases we were annotating which is why we weren’t getting all of the MVI annotations back for the variants we submitted in POST requests

Reply:
@onursumer maybe there is some bug which is causing the getQuery() field to be blanked out? If you can find and fix such a bug so that getQuery() will not return null for certain cases (perhaps occurring when a GET() request to annotation/genomic is first hit for a variant and then a POST() request follow subsequently) maybe fixing that bug would make this change unnecessary. (and maybe fix other problems we have not yet noticed) cc: @n1zea144

Reply:
@sheridancbio @ao508 I don't see `setQuery()` being manually called anywhere in our codebase. I think `query` is/was part of the My Variant Info response. Seems like at some point their model changed, so that we started to get `null` for this field for any variant that hasn't been already cached.

These types of bugs are difficult to detect unless we constantly monitor changes in external models. As long as we rely on live web services which frequently update their model, unfortunately we will continue having these types of issues.